concrete/src/Updater/Migrations/Migrations/Version20190509205043.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Concrete\Core\Updater\Migrations\Migrations;
+
+use Concrete\Core\Attribute\Category\CategoryService;
+use Concrete\Core\Attribute\TypeFactory;
+use Concrete\Core\Entity\Express\Control\Control;
+use Concrete\Core\Entity\Express\Control\PublicIdentifierControl;
+use Concrete\Core\Entity\Express\Entry;
+use Concrete\Core\Express\Entry\PublicIdentifierGenerator;
+use Concrete\Core\Updater\Migrations\AbstractMigration;
+use Doctrine\ORM\EntityManager;
+
+class Version20190509205043 extends AbstractMigration
+{
+
+    public function upgradeDatabase()
+    {
+        $generator = new PublicIdentifierGenerator();
+        $this->refreshEntities([
+            Entry::class,
+            PublicIdentifierControl::class,
+            Control::class,
+        ]);
+
+        // Now we have to populate this. Let's bust out of the entity manager for performance purposes.
+        $connection = $this->connection;
+        $this->connection->transactional(function() use ($connection, $generator) {
+            $r = $connection->executeQuery('select exEntryID from ExpressEntityEntries');
+            while ($row = $r->fetch()) {
+                $identifier = $generator->generate();
+                $connection->update(
+                    'ExpressEntityEntries',
+                    ['publicIdentifier' => $identifier],
+                    ['exEntryID' => $row['exEntryID']]
```

Comment:
Since this is now a repeatable migration, this migration can be executed more than once.
This code BTW overwrites the `publicIdentifier` column for each execution. Is this what we really want? Shouldn't we change the query from
```
select exEntryID from ExpressEntityEntries
```
to
```
select exEntryID from ExpressEntityEntries where publicIdentifier is null
```
?

Reply:
Yeah, that would be a good addition. We should only do this when that column is null.